src/daft-connect/src/translation/datatype.rs

```diff
@@ -112,3 +113,154 @@ pub fn to_spark_datatype(datatype: &DataType) -> spark_connect::DataType {
+
+    match kind {
+        Kind::Null(value) => {
+            ensure!(value.type_variation_reference == 0, type_variation_err);
+            Ok(DataType::Null)
+        }
+        Kind::Binary(value) => {
+            ensure!(value.type_variation_reference == 0, type_variation_err);
+            Ok(DataType::Binary)
+        }
+        Kind::Boolean(value) => {
+            ensure!(value.type_variation_reference == 0, type_variation_err);
+            Ok(DataType::Boolean)
+        }
+        Kind::Byte(value) => {
+            ensure!(value.type_variation_reference == 0, type_variation_err);
+            Ok(DataType::Int8)
+        }
+        Kind::Short(value) => {
+            ensure!(value.type_variation_reference == 0, type_variation_err);
+            Ok(DataType::Int16)
+        }
+        Kind::Integer(value) => {
+            ensure!(value.type_variation_reference == 0, type_variation_err);
+            Ok(DataType::Int32)
+        }
+        Kind::Long(value) => {
+            ensure!(value.type_variation_reference == 0, type_variation_err);
+            Ok(DataType::Int64)
+        }
+        Kind::Float(value) => {
+            ensure!(value.type_variation_reference == 0, type_variation_err);
+            Ok(DataType::Float32)
+        }
+        Kind::Double(value) => {
+            ensure!(value.type_variation_reference == 0, type_variation_err);
+            Ok(DataType::Float64)
+        }
+        Kind::Decimal(value) => {
+            ensure!(value.type_variation_reference == 0, type_variation_err);
+
+            let Some(precision) = value.precision else {
+                bail!("Decimal precision is required");
+            };
+
+            let Some(scale) = value.scale else {
+                bail!("Decimal scale is required");
+            };
+
+            let precision = usize::try_from(precision)
+                .wrap_err("Decimal precision must be a non-negative integer")?;
+
+            let scale =
+                usize::try_from(scale).wrap_err("Decimal scale must be a non-negative integer")?;
+
+            Ok(DataType::Decimal128(precision, scale))
+        }
+        Kind::String(value) => {
+            ensure!(value.type_variation_reference == 0, type_variation_err);
+            Ok(DataType::Utf8)
+        }
+        Kind::Char(value) => {
+            ensure!(value.type_variation_reference == 0, type_variation_err);
+            Ok(DataType::Utf8)
+        }
+        Kind::VarChar(value) => {
+            ensure!(value.type_variation_reference == 0, type_variation_err);
+            Ok(DataType::Utf8)
+        }
+        Kind::Date(value) => {
+            ensure!(value.type_variation_reference == 0, type_variation_err);
+            Ok(DataType::Date)
+        }
+        Kind::Timestamp(value) => {
+            ensure!(value.type_variation_reference == 0, type_variation_err);
+
+            // todo(?): is this correct?
+
+            Ok(DataType::Timestamp(TimeUnit::Microseconds, None))
```

Comment:
https://www.databricks.com/blog/2020/07/22/a-comprehensive-look-at-dates-and-timestamps-in-apache-spark-3-0.html

Sounds about right... Microseconds precision with no timezone info.

The timezones are interesting. It seems that Spark has a session-scoped timezone, but when writing data out to something like Parquet it will write it without timezone info (as UTC). I believe that the behavior you have here is correct though, given that Spark doesn't capture timezone information in the type itself but rather derives it from the current session.